src/get-oauth-authentication.ts

```diff
@@ -34,9 +34,17 @@ export async function getOAuthAuthentication(
     redirect_uri: options.redirectUrl
   };
 
+  const response = await request(route, parameters);
+
+  if (response.data.error !== undefined) {
+    throw new Error(
+      response.data.error_description || `Unknown error: ${response.data.error}`
+    );
+  }
+
   const {
     data: { access_token: token, scope }
-  } = await request(route, parameters);
+  } = response;
```

Comment:
Instead of throwing a custom error, I'd simulate the behavior that would occur if the server responded with a 400 status code. I've created https://github.com/octokit/request-error.js for cases such as this.

The code would look sth like this

```js
throw new RequestError(`${response.data.error_description} (${response.data.error})`, 400, {
  headers: response.headers,
  request: request.endpoint(route, parameters)
});
```

That way the error can be handled just like other errors such as server outages or request timeouts

Reply:
I implemented this change but used `response.status` (which will be `200`) rather than `400`... Because it doesn't sound like a good idea to fake the status code like that. Let me know if you disagree.